### PR TITLE
WPC-1099: Improved user first/lastname search

### PR DESCRIPTION
**Pull request by elias-summermatter**

### CODE REVIEW

**Comment by elias-summermatter** [comment]
@michael-scheurer @Itrulia please have a look :-)

File: libs/api/src/lib/graphql/user/user.queries.ts [hunk 1]
```diff
@@ -57,25 +57,159 @@ const createNameFilter = (filter: Partial<UserFilter>): Prisma.UserWhereInput =>
...
+const createUserNameFilter = (filter: Partial<UserFilter>): Prisma.UserWhereInput => {
+  const splitedString = filter.text.split(' ')
 
-const createTextFilter = (filter: Partial<UserFilter>): Prisma.UserWhereInput => {
-  if (filter?.text) {
+  if (splitedString.length === 1) {
     return {
       OR: [
         {
-          preferredName: {
```
Comment (penk85) [inline libs/api/src/lib/graphql/user/user.queries.ts:66]: Shouldn't the search still check `preferredName` in addition to first and last?

**Comment by elias-summermatter** [comment]
@Itrulia what do you mean by "every possibility"? Isn't that an bit overkill? Or do you have an idea how to achive "every possibility"?

File: libs/api/src/lib/graphql/user/user.queries.ts (hunk 1 above)
Comment (elias-summermatter) [reply libs/api/src/lib/graphql/user/user.queries.ts:66]: @penk85 ist still there just look at the new file without the changes there you will see that its still in the filter :-)

**Comment by elias-summermatter** [comment]
@penk85 @antkiewiczk can someone of you aprove this too? :-)

**Review by penk85 — APPROVED** [review]

**Pull request merged**